src/Product/ProductRepository.php

```diff
@@ -163,7 +164,8 @@ public function find_sync_ready_products( array $args = [], int $limit = - 1, in
 	 * @return array
 	 */
 	public function find_delete_product_ids( array $ids, int $limit = - 1, int $offset = 0 ): array {
-		$results = $this->find_by_ids( $ids, $limit, $offset );
+		$args    = [ 'status' => 'trash' ];
```

Comment:
This is now working for items that need to be deleted because they are in the trash. But for items that get deleted from Merchant Center for other reasons are not found/updated. For example I set a product to catalog visibility = hidden, which will trigger it to be deleted from the Merchant Center, but it's status remains "published".

Without any status it was previously querying the [default WP statuses](https://github.com/woocommerce/woocommerce/wiki/wc_get_products-and-WC_Product_Query#general) so it seems we should include `trash` with the default array.

```suggestion
		$args    = [ 'status' => [ 'draft', 'pending', 'private', 'publish', 'trash' ] ];
```

Reply:
Thanks for catching this, I've updated the issue #1785 so it will be handled in the next PR. Merging this PR first.